docs/contributing/coding-standards/component-options.md

```diff
@@ -4,204 +4,42 @@ Configuration options allow developers to pass additional information into a com
-  })
-}
-```
-
-### Validating a provided configuration against the schema
+## Preparing the component's JavaScript
 
-You can use the `validateConfig` function to ensure that a configuration object matches the schema format.
+If a component class needs configuration, it [can extend the `ConfigurableComponent` class](https://frontend.design-system.service.gov.uk/building-configurable-components/) that GOV.UK Frontend provides. This class accepts configuration from the following sources:
 
-If it doesn't, you can return a `ConfigError`.
+- [defaults defined as a `defaults` static property](https://frontend.design-system.service.gov.uk/building-configurable-components/#setting-a-default-configuration-for-your-component) on the component’s class
+- [configuration passed as second argument](https://frontend.design-system.service.gov.uk/building-configurable-components/#receiving-configuration-during-initialisation) to the component’s constructor or [`createAll`](https://frontend.design-system.service.gov.uk/building-your-own-javascript-components/#using-createall-with-your-components)
+- [data attributes on the root element](https://frontend.design-system.service.gov.uk/building-configurable-components/#receiving-configuration-from-data-attributes) of the component
 
-```mjs
-import { mergeConfigs, validateConfig, normaliseDataset } from '../../common/configuration.mjs'
-import { ConfigError } from '../../errors/index.mjs'
-import { GOVUKFrontendComponent } from '../../govuk-frontend-component.mjs'
+These are the same sources as those used by GOV.UK Frontend components. Configuration options from these sources are merged into a single configuration object by the `ConfigurableComponent` class. Note that:
 
-export class Accordion extends GOVUKFrontendComponent {
-  constructor($root, config = {}) {
-    super($root)
+- configuration options passed to the constructor will override any defaults
+- data attribute values have highest precedence and will override any configuration set elsewhere
 
-    this.config = mergeConfigs(
-      Accordion.defaults,
-      config,
-      normaliseDataset(Accordion, this.$root.dataset)
-    )
+Your component will then be able to:
 
-    // Check that the configuration provided is valid
-    const errors = validateConfig(Accordion.schema, this.config)
-    if (errors[0]) {
-      throw new ConfigError(formatErrorMessage(CharacterCount, errors[0]))
-    }
-  }
-}
-```
+[access the merged configuration](https://frontend.design-system.service.gov.uk/building-configurable-components/#accessing-the-component-s-configuration) using `this.config`
+use [the`Component` class](https://frontend.design-system.service.gov.uk/building-your-own-javascript-components/#using-the-component-class) features
 
 ## Adding a Nunjucks parameter
 
-Most, but not all, components support adding arbitrary attributes and values through the `attributes` parameter. This method is also more verbose compared to having a dedicated Nunjucks parameter.
```

Comment:
**suggestion** I think that bit is quite useful as it explains the two ways Nunjucks components can pass `data-*` attributes to the HTML for use by JavaScript components:
- via an `attributes` parameter allowing arbitrary `data-*` attributes 
- via specific Nunjucks options that get converted to `data-*` attributes

Reply:
Changed!